### PR TITLE
Running additional commands within tiab-entrypoint.sh

### PR DESCRIPTION
**Pull request by sadjy**
Did the following modifications:
- Added CLI option for additional commands run within `tiab-entrypoint.sh`: `-e / --extras`
- Changed dictionary interactions for python 3 compatibility
- Updated to kubernetes 9.0.0 client in requirements.txt
- Adapted teardown_cluster.py to the kube client update

Tested locally with the following steps:
1. ```python3 create_cluster.py -e "apt update && apt install unzip && wget http://search.maven.org/remotecontent\?filepath\=org/jacoco/jacoco/0.8.4/jacoco-0.8.4.zip -O jacoco.zip && mkdir /opt/jacoco && unzip jacoco.zip -d /opt/jacoco"  -i iotacafe/iri-dev -t 123abc -c /tmp/kube/config.yml -o /tmp/kube/output.yml -k /tmp/kube/kube.config -n <namespace> -d```
2. Notice the additional commands run logs in `output.yml` and `nc host port -v` connect to the pod.
3. ```python3 teardown_cluster.py -t 123abc -k /tmp/kube/kube.config -n <namespace>```
4. `nc host port -v` connection refused to the pod.

Also: 
- Added script `run_cmd.py` allowing to run commands on pod: ` ./run_cmd.py -k kube.config -o output.yml -n 'buildkite' -c "ls -al /"`

### CODE REVIEW

File: create_cluster.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 from getopt import getopt
 from jinja2 import Template
 from tempfile import TemporaryFile
+from functools import reduce
```
Comment (karimodm) [inline create_cluster.py]: This change is specific for Python3. `reduce` is not a built-in function anymore in python3. Should we change the shebang to explicitly run under python3 then?

File: create_cluster.py [hunk 2]
```diff
@@ -304,37 +310,33 @@ def deploy_monitoring(kubernetes_client, cluster):
...
-            if type(properties['java_options']) is str:
-                [e for e in iri_container['env'] if e['name'] == 'JAVA_OPTIONS'][0]['value'] = properties['java_options']
-            else:
-                raise RuntimeError('java_options for node %s is not a string' % node)
-        except KeyError as e:
-            if e[0] != 'java_options': raise e
+        java_options = properties.get('java_options')
+        if type(java_options) is str:
+            [e for e in iri_container['env'] if e['name'] == 'JAVA_OPTIONS'][0]['value'] = java_options
+        elif iri_args is not None:
```
Comment (karimodm) [inline create_cluster.py]: Paste mistake: `java_options`

File: run_cmd.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
```
Comment (karimodm) [inline run_cmd.py]: What's the goal of this file?
If you need a working example of `exec` with the kubernetes library you can look at the functions responsible for IXI modules packing, upload and extraction.

File: create_cluster.py (hunk 1 above)
Comment (sadjy) [reply create_cluster.py]: Good point. I didn't think it was necessary just yet. Should I?

File: create_cluster.py (hunk 2 above)
Comment (sadjy) [reply create_cluster.py]: Hum, which line please?

File: run_cmd.py (hunk 3 above)
Comment (sadjy) [reply run_cmd.py]: This file was a draft at an attempt not to use kubectl. I'll remove it.

File: create_cluster.py (hunk 1 above)
Comment (karimodm) [reply create_cluster.py]: Cool! If it still works for py2 I am fine without enforcing python3 usage.

**Pull request merged**